src/config.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as path from 'path';
+
+export interface ConfigOptions {
+  root?: string;
+  main?: string;
+  shell?: string;
+  entrypoints?: string[];
+}
+
+export class Config {
```

Comment:
To be a little more clear at he use sites of this type, let's call it something like `ProjectConfig`.